IdentityCore/src/oauth2/token/MSIDBoundRefreshToken.m

```diff
@@ -0,0 +1,154 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+#import "MSIDBoundRefreshToken.h"
+#import "MSIDBoundRefreshTokenCacheItem.h"
+
+@implementation MSIDBoundRefreshToken
+
+- (instancetype)initWithRefreshToken:(MSIDRefreshToken *)refreshToken
+                       boundDeviceId:(NSString *)boundDeviceId
+{
+    if (refreshToken && ![NSString msidIsStringNilOrBlank:boundDeviceId])
+    {
+        MSIDBoundRefreshToken *boundRefreshToken = [MSIDBoundRefreshToken new];
+        boundRefreshToken.refreshToken = refreshToken.refreshToken;
+        boundRefreshToken.boundDeviceId = boundDeviceId;
+        boundRefreshToken.familyId = refreshToken.familyId;
+        boundRefreshToken.storageEnvironment = refreshToken.storageEnvironment;
+        boundRefreshToken.environment = refreshToken.environment;
+        boundRefreshToken.realm = refreshToken.realm;
+        boundRefreshToken.clientId = refreshToken.clientId;
+        boundRefreshToken.additionalServerInfo = refreshToken.additionalServerInfo;
+        boundRefreshToken.accountIdentifier = refreshToken.accountIdentifier;
+        boundRefreshToken.speInfo = refreshToken.speInfo;
+        self = boundRefreshToken;
```

Comment:
Don't we need to do self = [super init] here?
Also, can we use the self.* or _vars syntax in this ctor instead of creating a temp object and assigning to self.

Reply:
Agreed, updated. I think I had modeled it based on how MSIDFamilyREfreshToken.m is written: https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/767b42b314d7f11d9ac89991be3caf6f498c7678/IdentityCore/src/oauth2/token/MSIDFamilyRefreshToken.m#L31C5-L49C6